[GeoMechanicsApplication] Create documentation for van Genuchten law (#14577)
KratosMultiphysics/Kratos: applications/GeoMechanicsApplication/custom_retention/README.md

| `VAN_GENUCHTEN_GN`                 | $n$         | Pore-size distribution index                              | $(0, \infty)$     |
| `VAN_GENUCHTEN_GL`                 | $l$         | Mualem pore-connectivity exponent                         | $(-\infty, \infty)$ — see note below |
| `MINIMUM_RELATIVE_PERMEABILITY`    | $k_{r,min}$ | Lower bound for $k_r$ (prevents zero permeability)        | $[0, 1]$          |

> **Note on `VAN_GENUCHTEN_GL` ($l$):** Although Mualem (1976) proposed $l = 0.5$ as a  general value, $l$ is a purely empirical pore-connectivity/tortuosity fitting parameter and **can be negative**. Schaap & Leij (2000) fitted the MVG model to hundreds of soils from the UNSODA database and found optimised values of $l$ ranging from approximately $-4$ to $+4$, with many fine-textured soils yielding $l < 0$. The HYDRUS software (Šimůnek, van Genuchten & Šejna, 2022) imposes **no lower bound** on $l$ and treats it as an unconstrained fitting parameter.

### Sign Convention for Fluid Pressure

Kratos uses the **soil mechanics sign convention** for fluid pressure $p$:

- $p > 0$  → suction (above phreatic level, unsaturated regime)  

> - $p \le 0$ → positive pore water pressure (at or below phreatic level, fully saturated)

Reviewer: Do you mean, negative water pressure?
To avoid confusion, maybe can be called "compression pore water pressure"

Author: corrected

#### Relative Permeability (Mualem–Van Genuchten)

$$
k_r = \max\left(
k_{r,\min},
S_e^{\,l}
\left[
1 - \left(1 - S_e^{1/m}\right)^m
\right]^2
\right)
$$

where $1/m = n/(n-1)$.

#### Bishop Effective Stress Coefficient

$$\chi = S_e$$

This is the simplest (and most common) choice for Bishop's parameter.

### Comparison with the Standard Van Genuchten (1980) Model

#### Standard formulation

Van Genuchten (1980) defined the soil water characteristic curve (SWCC) as:

$$\Theta \equiv \frac{\theta - \theta_r}{\theta_s - \theta_r}
  = \Bigl[1 + \bigl(\alpha\,|h|\bigr)^n\Bigr]^{-m}$$

with the Mualem constraint $m = 1 - 1/n$, $\alpha$ [1/length], $h$ = matric suction head (positive).

The Mualem (1976) relative permeability:

$$K_r = \Theta^l \Bigl[1 - \bigl(1-\Theta^{1/m}\bigr)^m\Bigr]^2$$

### Mapping to Kratos variables

| Standard symbol | Kratos variable                    | Relationship            |
|-----------------|------------------------------------|-------------------------|
| $\theta_s$      | `SATURATED_SATURATION`             | $S_s = \theta_s$        |
| $\theta_r$      | `RESIDUAL_SATURATION`              | $S_r = \theta_r$        |

> | $\alpha$        | `VAN_GENUCHTEN_AIR_ENTRY_PRESSURE` | $\alpha = 1/p_b$        |

Reviewer: Useually from experiments, we have GA. Then we calculate $P_b = \rho g /GA$. Adding explanation about GA can be useful here.

Author: corrected and added GA

# Retention laws

## Table of Contents

- [Van Genuchten Retention Law](#Van-Genuchten-Retention-Law)
- [Chapter 2](#chapter-2)

> - [Chapter 3](#chapter-3)

Reviewer: What are Chapter 2 and 3 ?

Author: for the future. I thought that other models shall be described as well.

  
## Van Genuchten Retention Law 

`VanGenuchtenLaw` is a C++ class in the **GeoMechanicsApplication** that implements the
Mualem–Van Genuchten (MVG) unsaturated soil model (Van Genuchten, 1980; Mualem, 1976).
It derives from the `RetentionLaw` base class and computes:

- Degree of saturation $S$
- Effective saturation $S_e$
- Derivative of saturation with respect to fluid pressure $\partial S / \partial p$
- Relative permeability $k_r$
- Bishop's effective stress coefficient $\chi$

### Material Parameters

| Kratos Variable                    | Symbol      | Description                                               | Constraints       |
|------------------------------------|-------------|-----------------------------------------------------------|-------------------|

> | `SATURATED_SATURATION`             | $S_s$       | Maximum (fully saturated) degree of saturation            | $[0, 1]$          |

Reviewer 2: ```suggestion
| `SATURATED_SATURATION`             | $S_s$ (or $\theta_s)      | Maximum (fully saturated) degree of saturation            | $[0, 1]$          |
```

Reviewer 2: To be in line with the notation in the van Genuchten paper. I see this was mentioned in the chapter mapping to kratos variables, but it would be nice to have that mapping before this chapter.

> | `RESIDUAL_SATURATION`              | $S_r$       | Residual degree of saturation                             | $[0, S_s)$        |

Reviewer 2: ```suggestion
| `RESIDUAL_SATURATION`              | $S_r$   (or $\theta_r$)    | Residual degree of saturation                             | $[0, S_s)$        |
```

> | `VAN_GENUCHTEN_AIR_ENTRY_PRESSURE` | $p_b$       | Air-entry (bubbling) pressure — inverse of VG scale $\alpha$ | $(0, \infty)$  |

Reviewer 2: Description is incorrect. I believe it is rho*g/alpha. @mnabideltares Can you confirm this?

Reviewer: p_b = rho*g/G_a 
here head is applied replacing of pressure, then  Pw=rho g h  , then Pb=rho g / alpha
This alpha is not same as the alpha in the other software

#### Degree of Saturation

$$S = \begin{cases}
S_s & \text{if } p \le 0 \\
S_r + (S_s - S_r)\bigl(1 + (p/p_b)^n\bigr)^{-m} & \text{if } p > 0
\end{cases}$$

#### Effective (Normalised) Saturation

$$S_e = \frac{S - S_r}{S_s - S_r} \in [0,\,1]$$

#### Derivative of Saturation with Respect to Fluid Pressure

$$\frac{\partial S}{\partial p} = \begin{cases}
0 & \text{if } p \le 0 \\
(S_s - S_r)\,(-m)\bigl(1+(p/p_b)^n\bigr)^{-m-1} \cdot n\,p_b^{-n}\,p^{n-1} & \text{if } p > 0
\end{cases}$$

> Note: The derivative is **negative** for $p > 0$ since saturation decreases as suction increases.

#### Relative Permeability (Mualem–Van Genuchten)

$$
k_r = \max\left(
k_{r,\min},
S_e^{\,l}
\left[
1 - \left(1 - S_e^{1/m}\right)^m
\right]^2
\right)
$$

where $1/m = n/(n-1)$.

#### Bishop Effective Stress Coefficient

$$\chi = S_e$$

This is the simplest (and most common) choice for Bishop's parameter.

### Comparison with the Standard Van Genuchten (1980) Model

#### Standard formulation

Van Genuchten (1980) defined the soil water characteristic curve (SWCC) as:

$$\Theta \equiv \frac{\theta - \theta_r}{\theta_s - \theta_r}

>   = \Bigl[1 + \bigl(\alpha\,|h|\bigr)^n\Bigr]^{-m}$$

Reviewer: Please remove the comma after alpha. It is multiplication

Author: it is a white space in vs code. removed

with the Mualem constraint $m = 1 - 1/n$, $\alpha$ [1/length], $h$ = matric suction head (positive).

The Mualem (1976) relative permeability:

$$K_r = \Theta^l \Bigl[1 - \bigl(1-\Theta^{1/m}\bigr)^m\Bigr]^2$$

> ### Mapping to Kratos variables

Reviewer 2: I think it would be nice to have this chapter earlier (when the material parameters are mentioned). Then the suggested changes at material parameters are not needed anymore.

Author: the mapping table is moved to the material parameters chapter

| Standard symbol | Kratos variable                    | Relationship            |
|-----------------|------------------------------------|-------------------------|
| $\theta_s$      | `SATURATED_SATURATION`             | $S_s = \theta_s$        |
| $\theta_r$      | `RESIDUAL_SATURATION`              | $S_r = \theta_r$        |

> | $\alpha$        | -                                  | $\alpha = 1/p_b$        |

Reviewer 2: also change the formulation here if it is incorrect.

> | $\alpha$        | -                                  | $\alpha = 1/p_b$        |

Reviewer: as alpha in the equation has the unit [1/length] , it doesn't match the unit here.
This alpha is not same as the one in "The other software". Here, pb=rho g/ alpha.

Author: corrected

> | $h_a$ or GA     | `VAN_GENUCHTEN_AIR_ENTRY_PRESSURE` | $\p_b=\rho_w g h_a$     |

Reviewer: What is h_a? 
here P_b = rho g / G_a

Author: h_a is a symbol for GA, removed

| $n$             | `VAN_GENUCHTEN_GN`                 | $n = \mathtt{gn}$       |
| $m$ (derived)   | —                                  | $m = 1 - 1/n$ (hardcoded) |
| $l$             | `VAN_GENUCHTEN_GL`                 | $l = \mathtt{gl}$       |

As a rule, GA is obtained from experiments that allows one to define $p_b$.

#### Equivalence

The Kratos formula is **mathematically identical** to the standard MVG model:

$$S = S_r + (S_s - S_r)\cdot\Theta, \qquad
  k_r = S_e^l\bigl[1-(1-S_e^{1/m})^m\bigr]^2$$

> with $\Theta = S_e$ evaluated at $p = \alpha^{-1}\,|h|$.

Reviewer: Again,  p is not equal as 1/alpha here, but Pb = rgo g / alpha

Author: corrected

#### Differences from the standard publication

| Aspect | Standard VG (1980) | Kratos `VanGenuchtenLaw` |
|--------|--------------------|--------------------------|
| Input variable | Matric suction head $h$ [m] | Fluid pressure $p$ [Pa or kPa] |

> | Scale parameter | $\alpha = 1/p_b$ | $p_b$ directly |

Reviewer: The same as above

Author: corrected

# Retention laws

## Table of Contents

- [Van Genuchten Retention Law](#Van-Genuchten-Retention-Law)
- [Chapter 2](#chapter-2)

> - [Chapter 3](#chapter-3)

Reviewer 3: In case you would like to describe the fully saturated law and the saturated below phreatic level law, I think that the order should be
1 fully saturated law
2 saturated below phreatic level law
3 Van Genuchten law
Otherwise remove the void chapter references.

| $\theta_s$      | `SATURATED_SATURATION`             | $S_s = \theta_s$        |
| $\theta_r$      | `RESIDUAL_SATURATION`              | $S_r = \theta_r$        |
| $\alpha$        | -                                  | $\alpha = \rho g/p_b$   |
| GA              | `VAN_GENUCHTEN_AIR_ENTRY_PRESSURE` | $p_b=\rho_w g GA$       |
| $n$             | `VAN_GENUCHTEN_GN`                 | $n = \mathtt{gn}$       |
| $m$ (derived)   | —                                  | $m = 1 - 1/n$ (hardcoded) |
| $l$             | `VAN_GENUCHTEN_GL`                 | $l = \mathtt{gl}$       |

> **Note on `VAN_GENUCHTEN_GL` ($l$):** Although Mualem (1976) proposed $l = 0.5$ as a  general value, $l$ is a purely empirical pore-connectivity/tortuosity fitting parameter and **can be negative**. Schaap & Leij (2000) fitted the MVG model to hundreds of soils from the UNSODA database and found optimised values of $l$ ranging from approximately $-4$ to $+4$, with many fine-textured soils yielding $l < 0$. The HYDRUS software (Šimůnek, van Genuchten & Šejna, 2022) imposes **no lower bound** on $l$ and treats it as an unconstrained fitting parameter.

### Sign Convention for Fluid Pressure

> Kratos uses the **soil mechanics sign convention** for fluid pressure $p$:

Reviewer 3: ```suggestion
Kratos uses the **tension positive sign convention** for fluid pressure $p$:
```
this is the opposite of the mentioned soil mechanics convention. The formulas below are correct.  Although I would rather speak of partially saturated than of unsaturated and I've never come across "compression pore water pressure". I you want to indicate more than "pore water pressure" I would write "compressive pore water pressure"

  = \Bigl[1 + \bigl(\alpha|h|\bigr)^n\Bigr]^{-m}$$

with the Mualem constraint $m = 1 - 1/n$, $\alpha$ [1/length], $h$ = matric suction head (positive).

The Mualem (1976) relative permeability:

$$K_r = \Theta^l \Bigl[1 - \bigl(1-\Theta^{1/m}\bigr)^m\Bigr]^2$$

#### Equivalence

The Kratos formula is **mathematically identical** to the standard MVG model:

$$S = S_r + (S_s - S_r)\cdot\Theta, \qquad
  k_r = S_e^l\bigl[1-(1-S_e^{1/m})^m\bigr]^2$$

with $\Theta = S_e$ evaluated at $p_b = \rho g /\alpha$.

#### Differences from the standard publication

| Aspect | Standard VG (1980) | Kratos `VanGenuchtenLaw` |
|--------|--------------------|--------------------------|
| Input variable | Matric suction head $h$ [m] | Fluid pressure $p$ [Pa or kPa] |
| Scale parameter | $\alpha = \rho g/p_b$ | $p_b$ directly |
| $m$ independent? | Optional (some software) | No — $m = 1 - 1/n$ is hardcoded |
| Saturated branch | Not defined separately | $S = S_s$ for $p \le 0$ |
| $S_s$ value | Usually $\theta_s = 1$ | Any value $\in (0,1]$ |
| Min permeability clamp | Not in original | Yes — `MINIMUM_RELATIVE_PERMEABILITY` |

### Known Limitations and Potential Issues

> 1. **Discontinuous derivative at $p = 0$**: `CalculateDerivativeOfSaturation` returns 0    for $p \le 0$ but is non-zero just above $p = 0$. This C⁰ discontinuity may hinder quadratic convergence of Newton iterations near the phreatic surface.

Reviewer 3: ```suggestion
1. **Discontinuous derivative at $p = 0$**: `CalculateDerivativeOfSaturation` returns 0    for $p \le 0$ but is non-zero just above $p = 0$. This C1 discontinuity may hinder quadratic convergence of Newton iterations near the phreatic surface.
```

From mathematics classes I think you would only speak about continuity ( the function is C0 continuous, but not C1 continuous )
Or are my memories from mathematics class wrong?

I think that hinder quadratic convergence is an understatement. A too low tangent may lead to divergence.

- Brinkgreve, R. B. J. et al. (2006). *PLAXIS Geotechnical Code for Soil and Rock Analyses.* Delft, Netherlands.
- Khalili, N. & Khabbaz, M. H. (1998). *A unique relationship for χ for the determination of the shear strength of unsaturated soils.* Géotechnique, 48(5):681–687.
- Bishop, A. W. (1959). *The principle of effective stress.* Teknisk Ukeblad, 106(39):859–863.
- Thom, C. S., Springman, S. M., Rouainia, M., van Esch, J. M. et al. (2017). *Numerical modelling of slope–vegetation–atmosphere interaction: an overview.* Quarterly Journal of Engineering Geology and Hydrogeology, 50(3):249–270. DOI: 10.1144/qjegh2016-079
- van Esch, J. M. (2012). *Modeling groundwater flow through dikes for real time stability assessment.* Proc. Computational Water Resources (CMWR), Urbana-Champaign, Illinois.
- Schaap, M. G. & Leij, F. J. (2000). *Improved prediction of unsaturated hydraulic conductivity with the Mualem–van Genuchten model.* Soil Sci. Soc. Am. J., 64(3):843–851.
  DOI: 10.2136/sssaj2000.643843x — reports optimised $l$ values ranging from $\approx -4$ to $+4$ across the UNSODA database; many fine-textured soils have $l < 0$.

> - Šimůnek, J., van Genuchten, M. Th., & Šejna, M. (2022). *The HYDRUS Software Package for Simulating the One-, Two-, and Three-Dimensional Movement of Water, Heat, and Multiple Solutes in Variably-Saturated Media, Technical Manual I, Version 5.0.* PC Progress, Prague — treats $l$ as an unconstrained real-valued fitting parameter.

Reviewer 3: Impressive list of references. Are we really using all of them? If we use a reference to the Plaxis manual, shouldn't that be a very recent one?
